lib/postgresql.js

```diff
@@ -766,6 +766,26 @@ PostgreSQL.prototype.toColumnValue = function(prop, val, isWhereClause) {
     });
   }
 
+  if (prop.type instanceof Array) {
+    // There is two possible cases for the type of "val" as well as two cases for dataType
+    const isArrayDataType = prop.postgresql &&
+                            prop.postgresql.dataType &&
+                            (prop.postgresql.dataType.indexOf('[]') > -1);
```

Comment:
There may be a more elegant way to detect this but I couldn't find one...

Reply:
@onaluf is `varchar[]` the only possible data type? or datatype in format `*[]` are all valid conditions?

If `varchar[]` is the only valid one, you can use 
```js
const isArrayDataType = prop.postgresql &&
prop.postgresql.dataType === 'varchar[]';
```
instead.